python/paddle/distributed/fleet/base/fleet_base.py

```diff
@@ -253,6 +253,40 @@ def init(self, role_maker=None, is_collective=False, strategy=None):
                 warnings.warn(
                     "The dygraph hybrid parallel environment has been initialized."
                 )
+        elif self._is_collective:
+            use_sharding = self._user_defined_strategy.sharding
+
+            # global group
+            global_rank = self.worker_index()
+            global_world_size = self.worker_num()
+            # NOTE(wangxi): see sharding_optimizer
```

Comment:
or we could explain the parallel2id mapping here again?

Reply:
pre-assign ring ids
            mp: 0
            sharding: 1
            pure-dp: 2
            global: 3
            pp: >= 20
        if one parallelism is not enable: -1
        and only support parallelism hierarchy: mp --> sharding --> pp --> dp      


done